src/NuGetGallery.Core/Auditing/FederatedCredentialPolicyAuditRecord.cs

```
@@ -224,45 +224,37 @@ public static FederatedCredentialPolicyAuditRecord FirstUseUpdate(
         }
 
         public static FederatedCredentialPolicyAuditRecord BadRequest(
-            FederatedCredentialType type,
-            User createdBy,
-            User packageOwner,
-            string criteria,
-            string errorMessage)
+            FederatedCredentialPolicy policy, string errorMessage)
         {
             return new FederatedCredentialPolicyAuditRecord(
                 AuditedFederatedCredentialPolicyAction.BadRequest,
-                null, // no DB policy key
-                type,
-                createdBy,
-                packageOwner,
-                criteria,
+                policy.Key,
+                policy.Type,
+                policy.CreatedBy,
+                policy.PackageOwner,
+                policy.Criteria,
                 success: false,
                 federatedCredential: null,
                 apiKeyCredentials: [],
                 externalCredential: null,
-                errorMessage: errorMessage);
+                errorMessage: errorMessage ?? throw new ArgumentNullException(nameof(errorMessage)));
```

Do we really need to throw an exception for missing error message here? This might cause 400 response replaced with 500. Is that really expected?